tests/drivers/uart/uart_basic_api/testcase.yaml

```diff
@@ -47,3 +47,13 @@ tests:
     filter: CONFIG_UART_CONSOLE
     depends_on: usb_device
     harness: keyboard
+  drivers.uart.basic_api.shell_ke17z9_uart:
```

Comment:
why is the extra test case needed?

Reply:
The frdm_ke17z512 has two communication interfaces, LPUART and UART(SCI). By default, the LPUART driver is used, and the "zephyr, console "and "zephyr,shell-uart" in dts use LPUART. So uart_basic_api tests LPUART by default, and these additional configurations are needed to test UAR drivers.